File: Dynamic_Programming_Old/LCS/lcs_recursion.js
function lcs(a, b, i, j) {
  if (a[i] === undefined || b[j] === undefined) {
    return 0;
  } else if (a[i] === b[j]) {
    return 1 + lcs(a, b, i + 1, j + 1);
  } else {
    return Math.max(lcs(a, b, i + 1, j), lcs(a, b, i, j + 1));
  }
}

console.log(lcs("kaium", "aium", 0, 0));
// a = "kaium";
// console.log(a[5]=== undefined);

File: Dynamic_Programming_Old/LCS/lcs_recursion.cpp
#include<bits/stdc++.h>
using namespace std;

int lcs(string a, string b, int i = 0, int j = 0) {
    if (a[i] == '\0' || b[j] == '\0')
        return 0;
    else if (a[i] == b[j])
        return 1 + lcs(a, b, i + 1, j + 1);
    else
        return max(lcs(a, b, i + 1, j), lcs(a, b, i, j + 1));
}

int main()
{
    cout << lcs("kaimu", "aiu") << endl;

    return 0;
}